### PR TITLE
Fix rare infinite loop

### PR DESCRIPTION
**Pull request by RhettAtBlack**
* Update Oct 27, 2017 by Rhett Anderson
 * Fix infinite loop that happens when an image that's too large to fit spills out into a second sheet

Details:
Spriter tries various size of maximum width to try to find the best power-of-2 packing. In a couple of very rare cases, one of the maxwidth values we try is smaller than the width of an image we're trying to put on it. This causes an infinite loop during the build. This fix bails out in such a case. This does not hurt the packing.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,6 +7,6 @@
   "dependencies": {
     "bluebird": "^2.10.2",
     "graceful-fs": "^4.1.2",
-    "jimp": "git+https://github.com/mgh/jimp.git"
+    "jimp": "^0.2.27"
```
Comment (bengarney) [inline package.json:10]: Is this an explicit version reference or a range? (ie > 0.2.27)

File: layout.js [hunk 2]
```diff
@@ -1,144 +1,289 @@
+
```
Comment (bengarney) [inline layout.js:1]: Is there any way we can test this to confirm it's working right?
Does it copy pixels always or just at end?

**Comment by heilhead** [comment]
@RhettAtBlack @bengarney Why are we merging this into `master`? I see some pretty old commits - I thought we support only `es6` branch these days?

Context: I'm duplicating these changes in `coretech_frontend` so need a clear picture of what's changed.

**Comment by RhettAtBlack** [comment]
Yes, my mistake. Will redo for es6.

**Pull request closed without merging**